src/Client/ClientBase.cpp

```diff
@@ -2160,6 +2169,7 @@ void ClientBase::processParsedSingleQuery(
     cancelled_printed = false;
     client_exception.reset();
     server_exception.reset();
+    client_context->setInsertionTable(StorageID::createEmpty());
```

Comment:
Probably we just need to create copy of the client context for each processed query

Reply:
Yes, that would be much prettier. Still, your solution should work. Could you add a TODO comment if you are not planning to do it in this PR?